assets/javascripts/discourse/components/realtime-validation-settings.js.es6

```diff
@@ -0,0 +1,35 @@
+import Component from "@ember/component";
+import EmberObject from "@ember/object";
+import { cloneJSON } from "discourse-common/lib/object";
+import Category from "discourse/models/category";
+
+export default Component.extend({
+    init(){
+        this._super(...arguments);
+        if (!this.validations) return;
+
+        if (!this.field.validations) {
+            const validations = {};
+            this.validations.forEach((validation) => {
+                validations[validation] = {};
+            });
+
+            this.set('field.validations', EmberObject.create(validations));
+        }
+
+        const validationBuffer = cloneJSON(this.get('field.validations'));
+        let bufferCategories;
+        if( validationBuffer.similar_topics && (bufferCategories = validationBuffer.similar_topics.categories)) {
```

Comment:
@fzngagan I'm not entirely clear on the role the buffer is playing here. Perhaps you can clarify this for me on our call.